### PR TITLE
update the translation of `Array.pop()`

### PR DESCRIPTION
**Pull request by qibing147147**
### Description

@JasonLamv-t I have update the translation of pop.I modified and translated it according to the English document
### Related issues and pull requests
[[zh-cn] The proofreading and update plan of Javascript.Array #12761](https://github.com/mdn/translated-content/issues/12761)

### CODE REVIEW

**Comment by JasonLamv-t** [comment]
👍 Thanks for your contribution! I already add this PR to the progress list.

File: files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md [hunk 1]
```diff
@@ -21,11 +21,13 @@ pop()
 
 ## 描述
 
-`pop` 方法从一个数组中删除并返回最后一个元素。
+`pop()` 方法从一个数组中删除并返回最后一个元素。
 
-`pop` 方法有意具有通用性。该方法和 {{jsxref("Function.call", "call()")}} 或 {{jsxref("Function.apply", "apply()")}} 一起使用时，可应用在类似数组的对象上。`pop` 方法根据 `length` 属性来确定最后一个元素的位置。如果不包含 `length` 属性或 `length` 属性不能被转成一个数值，会将 `length` 置为 0，并返回 `undefined`。
+{{jsxref("Array.prototype.shift()")}} 和 `pop()`有一样的行为, 但是他是作用在数组的第一个元素上的。
```
Comment (coolzjy) [inline files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md:26]:
```suggestion
{{jsxref("Array.prototype.shift()")}} 和 `pop()` 有一样的行为，但是它是作用在数组的第一个元素上的。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md [hunk 2]
```diff
@@ -21,11 +21,13 @@ pop()
...
 
-如果你在一个空数组上调用 `pop()`，它将返回 {{jsxref("undefined")}}。
+`pop()`方法是有副作用的。其改变了`this`的长度和内容。如果你想要`this`不变，但是返回一个新的最后一个元素被移除的数组，你可以使用[`arr.slice(0, -1)`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice)来代替。
```
Comment (coolzjy) [inline files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md:28]:
```suggestion
`pop()` 方法是有副作用的。其改变了 `this` 的长度和内容。如果你想要 `this` 不变，但是返回一个新的最后一个元素被移除的数组，你可以使用 [`arr.slice(0, -1)`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) 来代替。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md [hunk 3]
```diff
@@ -21,11 +21,13 @@ pop()
...
+
+`pop()`方法具有[通用性](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#generic_array_methods)。其只期望`this`上有`length`属性或者整数键属性。虽然字符串也是类数组对象，但是由于其不能被修改，所以`pop()`方法并不能应用在字符串上。
```
Comment (coolzjy) [inline files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md:30]:
```suggestion
`pop()` 方法是[通用的](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#generic_array_methods)。它只期望 `this` 值具有 `length` 属性和整数键属性。虽然字符串也是类数组对象，但是由于其不能被修改，所以 `pop()` 方法并不能应用在字符串上。
```
与其他通用数组方法统一表述。

File: files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md [hunk 4]
```diff
@@ -43,6 +45,61 @@ console.log(myFish); // ['angel', 'clown', 'mandarin']
 console.log(popped); // 'sturgeon'
 ```
 
+### 在非数组对象上调用pop()
```
Comment (coolzjy) [inline files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md:48]:
```suggestion
### 在非数组对象上调用 pop()
```

File: files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md [hunk 5]
```diff
@@ -43,6 +45,61 @@ console.log(myFish); // ['angel', 'clown', 'mandarin']
...
+
+`pop()`方法会读取`this`上的`length`属性。如果[规范化的length属性](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#normalization_of_the_length_property)为0，`length`会被再次设置为0（或者之前可能是负数或者`undefined`）。否则，[删除](/en-US/docs/Web/JavaScript/Reference/Operators/delete)并返回`length-1`的属性。
```
Comment (coolzjy) [inline files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md:50]:
```suggestion
`pop()` 方法会读取 `this` 上的 `length` 属性。如果[规范化的 length 属性](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#normalization_of_the_length_property)为 0，`length`会被再次设置为 0（鉴于之前可能是负数或者 `undefined`）。否则，返回并[删除](/en-US/docs/Web/JavaScript/Reference/Operators/delete)位于 `length-1` 处的属性。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md [hunk 6]
```diff
@@ -43,6 +45,61 @@ console.log(myFish); // ['angel', 'clown', 'mandarin']
...
+  unrelated: "foo",
+  2: 4,
+};
+console.log(Array.prototype.pop.call(arrayLike));
+// 4
+console.log(arrayLike);
+// { length: 2, unrelated: 'foo' }
+
+const plainObj = {};
+// 没有length属性，所以长度为0
```
Comment (coolzjy) [inline files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md:64]:
```suggestion
// 没有 length 属性，所以长度为 0
```

File: files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md [hunk 7]
```diff
@@ -43,6 +45,61 @@ console.log(myFish); // ['angel', 'clown', 'mandarin']
...
+Array.prototype.pop.call(plainObj);
+console.log(plainObj);
+// { length: 0 }
+```
+
+### 以类数组的方式使用对象
+
+`push`和`pop`方法是通用的，我们可以利用这一点来编写更灵活的代码--如以下示例所示。
```
Comment (coolzjy) [inline files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md:72]:
```suggestion
`push` 和 `pop` 方法是通用的，我们可以利用这一点来编写更灵活的代码——如以下示例所示。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md [hunk 8]
```diff
@@ -43,6 +45,61 @@ console.log(myFish); // ['angel', 'clown', 'mandarin']
...
+
+请注意，在此示例中，我们不会创建数组来存储对象集合。 相反，我们将集合存储在对象本身上，并在 `Array.prototype.push` 和 `Array.prototype.pop` 上使用 `call` 来欺骗这些方法，让它们认为我们正在处理一个数组。
```
Comment (coolzjy) [inline files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md:74]:
```suggestion
请注意，在此示例中，我们不会创建数组来存储对象集合。相反，我们将集合存储在对象本身上，并在 `Array.prototype.push` 和 `Array.prototype.pop` 上使用 `call` 来欺骗这些方法，让它们认为我们正在处理一个数组。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md [hunk 9]
```diff
@@ -43,6 +45,61 @@ console.log(myFish); // ['angel', 'clown', 'mandarin']
...
+
+```js
+const collection = {
+  length: 0,
+  addElements(...elements) {
+    // 每次添加元素时
+    // obj.length都会自动增加
+
+    // 返回push方法的返回值
+    // 为length属性的新值
```
Comment (coolzjy) [inline files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md:84]:
```suggestion
    // obj.length 都会自动增加

    // 返回 push 方法的返回值，即 length 属性的新值
```

File: files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md [hunk 10]
```diff
@@ -43,6 +45,61 @@ console.log(myFish); // ['angel', 'clown', 'mandarin']
...
+    return [].push.call(this, ...elements);
+  },
+  removeElement() {
+    // 每次移除元素时
+    // obj.length都会自动减少
+
+    // 返回pop方法的返回值
+    // 为被移除的元素
```
Comment (coolzjy) [inline files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md:92]:
```suggestion
    // obj.length 都会自动减少

    // 返回 pop 方法的返回值，即被移除的元素
```

**Comment by coolzjy** [comment]
See also 部分更新没有同步

**Comment by qibing147147** [comment]
> See also 部分更新没有同步

好的。谢谢您。我会再改一下

**Comment by qibing147147** [comment]
> See also 部分更新没有同步

您好 我已经更新了我的提交，麻烦您有空的时候再帮忙review一下

File: files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md [hunk 11]
```diff
@@ -21,11 +21,13 @@ pop()
...
-`pop` 方法从一个数组中删除并返回最后一个元素。
+`pop()` 方法从一个数组中删除并返回最后一个元素。
 
-`pop` 方法有意具有通用性。该方法和 {{jsxref("Function.call", "call()")}} 或 {{jsxref("Function.apply", "apply()")}} 一起使用时，可应用在类似数组的对象上。`pop` 方法根据 `length` 属性来确定最后一个元素的位置。如果不包含 `length` 属性或 `length` 属性不能被转成一个数值，会将 `length` 置为 0，并返回 `undefined`。
+{{jsxref("Array.prototype.shift()")}} 和 `pop()` 有一样的行为，但是它是作用在数组的第一个元素上的。
 
-如果你在一个空数组上调用 `pop()`，它将返回 {{jsxref("undefined")}}。
+`pop()` 方法是有副作用的。其改变了 `this` 的长度和内容。如果你想要 `this` 不变，但是返回一个新的最后一个元素被移除的数组，你可以使用 [`arr.slice(0, -1)`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) 来代替。
+
+`pop()` 方法是[通用的](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#generic_array_methods)。它只期望 `this` 值具有 `length` 属性和整数键属性。虽然字符串也是类数组对象，但是由于其不能被修改，所以 `pop()` 方法并不能应用在字符串上。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md:30]:
```suggestion
`pop()` 方法是[通用的](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#通用数组方法)。它只期望 `this` 值具有 `length` 属性和整数键属性。虽然字符串也是类数组对象，但是由于其不能被修改，所以 `pop()` 方法并不能应用在字符串上。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md [hunk 12]
```diff
@@ -43,6 +45,59 @@ console.log(myFish); // ['angel', 'clown', 'mandarin']
 console.log(popped); // 'sturgeon'
 ```
 
+### 在非数组对象上调用 pop()
+
+`pop()` 方法会读取 `this` 上的 `length` 属性。如果[规范化的 length 属性](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#normalization_of_the_length_property)为 0，`length`会被再次设置为 0（鉴于之前可能是负数或者 `undefined`）。否则，返回并[删除](/en-US/docs/Web/JavaScript/Reference/Operators/delete)位于 `length-1` 处的属性。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md:50]:
```suggestion
`pop()` 方法会读取 `this` 上的 `length` 属性。如果[规范化的 length 属性](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#长度属性的规范化)为 0，`length` 会被再次设置为 0（鉴于之前可能是负数或者 `undefined`）。否则，返回并[删除](/zh-CN/docs/Web/JavaScript/Reference/Operators/delete)位于 `length-1` 处的属性。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md [hunk 13]
```diff
@@ -53,6 +108,8 @@ console.log(popped); // 'sturgeon'
 
 ## 参见
 
+- [索引集合](/en-US/docs/Web/JavaScript/Guide/Indexed_collections)
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md:111]:
```suggestion
- [索引集合类](/zh-CN/docs/Web/JavaScript/Guide/Indexed_collections)
```

**Review by JasonLamv-t — COMMENTED** [review]
Some suggestions

File: files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md [hunk 14]
```diff
@@ -21,11 +21,13 @@ pop()
 
 ## 描述
 
-`pop` 方法从一个数组中删除并返回最后一个元素。
+`pop()` 方法从一个数组中删除并返回最后一个元素。
 
-`pop` 方法有意具有通用性。该方法和 {{jsxref("Function.call", "call()")}} 或 {{jsxref("Function.apply", "apply()")}} 一起使用时，可应用在类似数组的对象上。`pop` 方法根据 `length` 属性来确定最后一个元素的位置。如果不包含 `length` 属性或 `length` 属性不能被转成一个数值，会将 `length` 置为 0，并返回 `undefined`。
+{{jsxref("Array.prototype.shift()")}} 和 `pop()` 有一样的行为，但是它是作用在数组的第一个元素上的。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md:26]:
```suggestion
{{jsxref("Array.prototype.shift()")}} 和 `pop()` 有类似的行为，但是它是作用在数组的第一个元素上的。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md [hunk 15]
```diff
@@ -21,11 +21,13 @@ pop()
...
 
-如果你在一个空数组上调用 `pop()`，它将返回 {{jsxref("undefined")}}。
+`pop()` 方法是有副作用的。其改变了 `this` 的长度和内容。如果你想要 `this` 不变，但是返回一个新的最后一个元素被移除的数组，你可以使用 [`arr.slice(0, -1)`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) 来代替。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/array/pop/index.md:28]:
```suggestion
`pop()` 是修改方法。其改变了 `this` 的长度和内容。如果你想要 `this` 不变，但是返回一个新的最后一个元素被移除的数组，你可以使用 [`arr.slice(0, -1)`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) 来代替。
```

**Review by yin1999 — APPROVED** [review]

**Comment by yin1999** [comment]
Thanks @qibing147147, @JasonLamv-t, and  @coolzjy!

**Pull request merged**